### ISSUE TITLE
vcd2fsdb failed

### ISSUE THREAD

**Issue by yangguangguang** [steveicarus/iverilog#883]
Hi:

  I`m running the [verilog-axi](https://github.com/alexforencich/verilog-axi) testbenches and use the following configuration to generate the fst file：

```
ifeq ($(WAVES), 1)
VERILOG_SOURCES += iverilog_dump.v
COMPILE_ARGS += -s iverilog_dump
endif

iverilog_dump.v:
	echo 'module iverilog_dump();' > $@
	echo 'initial begin' >> $@
	echo '    $$dumpfile("$(TOPLEVEL).fst");' >> $@
	echo '    $$dumpvars(0, $(TOPLEVEL));' >> $@
	echo 'end' >> $@
	echo 'endmodule' >> $@
  
```

and then I use 'fst2vcd' command converts fst file to vcd, I can use GTKWave to view all waveforms. But when I use 'vcd2fsdb' command converts vcd file to fsdb, and use Verdi to view waveforms, then I can only view the  three layers waveforms of the fsdb file.The warning of the Verdi is that failed to add certain signals to waveform because they cannot be found in FSDB.

what could be the cause of the problem？Thanks

**Comment by martinwhitaker** [comment]
Do you get different results if you let vvp generate the VCD file itself?

**Comment by yangguangguang** [comment]
I tried a new method and it worked：
1: use the 'fsdbdebug' tools to view the *.fsdb to ensure that the waveform of the signal is present.
2: open verdi like this:
![111](https://user-images.githubusercontent.com/50956802/224905404-c210deec-6324-4dde-bf07-0234ec2737bf.jpg)
I found that a '\\' prefix is ​​added in front of the m_ifaces[0], but in the Hierarchy Browser Pane, I can't see the '\\' prefix added in front of m_ifaces[0]. in the *.fsdb, I found that m_ifaces[0] is recorded like this:
![222](https://user-images.githubusercontent.com/50956802/224908132-88d0955c-e789-4bca-be89-8dcb1079d575.jpg)
Whenever I add a signal waveform through the Hierarchy Browser Pane, verdi prompts an error message("Failed to add certain signals to waveform because they cannot be found in FSDB"). When I checked the original vcd file, I found no abnormal phenomenon:
![333](https://user-images.githubusercontent.com/50956802/224909603-bb2d4251-0344-4dd1-b74c-089bbcf1f46f.jpg)
rtl uses 'generate for' to create instantiations of modules, not sure if this is the reason, thanks a lot!
